File: maze-walker/typescript/tests/WalkerBuilder.ts
import { type Maze } from '../src/Maze.js';
import { Walker } from '../src/Walker.js';
import { MazeBuilder } from './MazeBuilder.js';

export class WalkerBuilder {
  private maze: Maze = new MazeBuilder().build();

  withMaze(maze: Maze): this {
    this.maze = maze;
    return this;
  }

  build(): Walker {
    return new Walker(this.maze);
  }
}

File: maze-walker/python/tests/walker_builder.py
from __future__ import annotations

from maze_walker import Maze, Walker
from tests.maze_builder import MazeBuilder


class WalkerBuilder:
    def __init__(self) -> None:
        self._maze: Maze = MazeBuilder().build()

    def with_maze(self, maze: Maze) -> WalkerBuilder:
        self._maze = maze
        return self

    def build(self) -> Walker:
        return Walker(self._maze)
